test/api/role.spec.js

```diff
@@ -0,0 +1,147 @@
+            expect(response.status).to.equal(201);
+            done();
+          });
+      });
+  });
+
+  beforeEach((done) => {
+    model.Role.create(regularRoleParams)
+      .then((regularRole) => {
+        role = regularRole;
+        done();
+      });
+  });
+
+  afterEach(() => {
+    return model.Role.destroy({ where: { id: role.id } });
+  });
+
+  after(() => {
+    return model.sequelize.sync({ force: true });
+  });
+
+  describe('REQUESTS', () => {
+    describe('POST: (/roles) - CREATE ROLE', () => {
+      it('should create a role when required field is valid', (done) => {
+        const newRole = { title: 'super duper Admin' };
+        request.post('/roles')
+          .set({ Authorization: token })
+          .send(newRole)
+          .end((error, response) => {
+            expect(response.status).to.equal(201);
+            expect(response.body.title).to.equal(newRole.title);
+            done();
+          });
+      });
+      it('should not create a role when required field is invalid', (done) => {
+        const newRole = { name: 'guest' };
+        request.post('/roles')
+          .set({ Authorization: token })
+          .send(newRole)
+          .expect(400, done);
+      });
+    });
+    describe('GET: (/roles)', () => {
+      it('should not return roles where no token is provided', (done) => {
+        request.get('/roles')
+          .expect(401, done);
+      });
+      it('should not return roles where token is invalid', (done) => {
+        request.get('/roles')
+          .set({ Authorization: 'jbugubhbhkbkbkb' })
+          .expect(401, done);
+      });
+      it('should return roles where token is valid', (done) => {
+        request.get('/roles')
+          .set({ Authorization: token })
+          .end((error, response) => {
+            expect(response.status).to.equal(200);
+            // eslint-disable-next-line no-unused-expressions
+            expect(Array.isArray(response.body)).to.be.true;
+            expect(response.body.length).to.be.greaterThan(0);
+            done();
+          });
+      });
+    });
+
+    describe('GET: (/roles/:id) - GET ROLE', () => {
+      it('should not return the role when supplied invalid id', (done) => {
+        request.get('/roles/999999')
+          .set({ Authorization: token })
+          .end((error, response) => {
+            expect(response.status).to.equal(404);
+            done();
+          });
+      });
+      it('should return the role when valid id is provided', (done) => {
+        request.get(`/roles/${role.id}`)
+          .set({ Authorization: token })
+          .end((error, response) => {
+            expect(response.status).to.equal(200);
+            done();
+          });
+      });
+    });
+
+    describe('PUT: (/roles/:id) - EDIT ROLE', () => {
+      it('should not perform edit if wrong id is supplied', (done) => {
+        const fieldsToUpdate = { title: 'super Admin' };
+        request.put('/roles/45')
```

Comment:
Although this is a test environment, however you may want to set the id to a higher value `like: 999999` so the test doesn't readily fail if test specs are increased a little bit.